### DISCUSSION TITLE
Share your use case

### DISCUSSION THREAD

**Discussion by ziflex** [MontFerret/ferret#661] [Show and tell]
Hello folks!

I would like to ask you to share your use cases of using Ferret in order to better understand in which direction the project should move.

**Comment by ziflex** [comment]
Thanks for sharing!
Is there something you would want to add to Ferret in the future?

**Comment by ziflex** [comment]
Speaking of GUI, there is [Dashboard UI](https://github.com/MontFerret/ferret-server-ui) for [Ferret Server](https://github.com/MontFerret/ferret-server), it's still WIP, but you are welcome to contribute.

**Comment by ap0** [comment]
We are using it to scrape home showing data for our company’s real estate listings because they provide no API to access it. Overall it’s been a pretty solid solution save for issues where the sites we scrape are slow to load or have issues due to lots of SSO redirection. A nice to have feature would be something like “navigate to a URL and wait for it to get to this other URL after a series of redirections”

**Comment by ziflex** [comment]
@ap0 thank you for sharing!
Hmm, this is definitely something we need to think about. Thanks for the idea!

**Comment by ziflex** [comment]
@ap0 how did you integrate it? Are you using it via script as a CLI or embedded into your scraping system?

**Comment by ap0** [comment]
> @ap0 how did you integrate it? Are you using it via script as a CLI or embedded into your scraping system?

We embed it into the scraping system.

**Comment by hubitor** [comment]
I would like to extract data from a each row of a given listing. Extract the links if they exist in the rows, follow these links, extract content form these pages. This must be done for all pages of the listing.

I personally would be very happy for any direction that would make scraping fast and mostly automatized.

**Comment by suntong** [comment]
**Structured scrapping**

The following is the approach that I had been needing the most (adapted from https://github.com/andrew-d/goscrape):

The architecture of scraping is roughly as follows:

- A single request to start scraping (from a given URL).
- Each scrape consists of some number of *pages*.
- Inside each page, there's 1 or more *blocks* - some logical method of splitting
  up a page into subcomponents.  By default, there's a single block that consists
	of the pages's `<body>` element, but you can change this fairly easily.
- Inside each block, you define some number of *pieces* of data that you wish
  to extract.  Each piece consists of a name, a selector, and what data to
	extract from the current block.

Almost every single task that I need to do falls within such description. 

Like in https://www.montferret.dev/cookbook/google-search/, but need an outer loop to walk page by page until limit is reached. 
**Update:** just like in https://www.montferret.dev/cookbook/pagination/, but so far the first two examples that I tried there are broken.
